aws/resource_aws_ec2_client_vpn_authorization_rule_test.go

```diff
@@ -0,0 +1,326 @@
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+		result, err := findClientVpnAuthorizationRuleByID(conn, rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("error reading Client VPN authorization rule (%s): %w", rs.Primary.ID, err)
+		}
+
+		if result != nil || len(result.AuthorizationRules) == 1 || result.AuthorizationRules[0] != nil {
+			*assoc = *result.AuthorizationRules[0]
+			return nil
+		}
+
+		return fmt.Errorf("Client VPN network association (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccEc2ClientVpnAuthorizationRuleConfigBasic(rName string) string {
+	return testAccEc2ClientVpnVpcComposeConfig(rName, 1, `
+resource "aws_ec2_client_vpn_authorization_rule" "test" {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.test.id
+  target_network_cidr    = aws_subnet.test[0].cidr_block
+  authorize_all_groups   = true
+}
+`)
+}
+
+func testAccEc2ClientVpnAuthorizationRuleConfigGroups(rName string, groupNames map[string]string) string {
+	var b strings.Builder
+	for k, v := range groupNames {
+		fmt.Fprintf(&b, `
+resource "aws_ec2_client_vpn_authorization_rule" %[1]q {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.test.id
+  target_network_cidr    = aws_subnet.test[0].cidr_block
+  access_group_id        = %[2]q
+}
+`, k, v)
+	}
+
+	return testAccEc2ClientVpnVpcComposeConfig(rName, 1, b.String())
+}
+
+func testAccEc2ClientVpnAuthorizationRuleConfigSubnets(rName string, subnetCount int, groupNames map[string]int) string {
+	var b strings.Builder
+	for k, v := range groupNames {
+		fmt.Fprintf(&b, `
+resource "aws_ec2_client_vpn_authorization_rule" %[1]q {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.test.id
+  target_network_cidr    = aws_subnet.test[%[2]d].cidr_block
+  authorize_all_groups   = true
+}
+`, k, v)
+	}
+
+	return testAccEc2ClientVpnVpcComposeConfig(rName, subnetCount, b.String())
+}
+
+func testAccEc2ClientVpnVpcBase(rName string, subnetCount int) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "available" {
+  # InvalidParameterValue: AZ us-west-2d is not currently supported. Please choose another az in this region
+  exclude_zone_ids = ["usw2-az4"]
+  state            = "available"
+
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = "terraform-testacc-subnet-%[1]s"
+  }
+}
+
+resource "aws_subnet" "test" {
+  count                   = %[2]d
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  cidr_block              = cidrsubnet(aws_vpc.test.cidr_block, 8, count.index)
+  vpc_id                  = aws_vpc.test.id
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "tf-acc-subnet-%[1]s"
+  }
+}
+`, rName, subnetCount)
+}
+
+func testAccEc2ClientVpnVpcComposeConfig(rName string, subnetCount int, config ...string) string {
```

Comment:
Similar to the above comment, this feels harder to follow than our typical test configuration patterns.